source/features/add-pr-branches.js

```diff
@@ -0,0 +1,96 @@
+import {h} from 'dom-chef';
+import select from 'select-dom';
+import * as api from '../libs/api';
+import {getOwnerAndRepo} from '../libs/page-detect';
+
+function buildQuery(owner, repo, numbers) {
+	let query = `
+	{
+		repository(owner: "${owner}", name: "${repo}") {`;
+
+	for (const number of numbers) {
+		query += `
+			${number}: pullRequest(number: ${number.replace('issue_', '')}) {
+				baseRef {
+					id
+				}
+				baseRefName
+
+				headRef {
+					id
+				}
+				headRefName
+				headRepository {
+					url
+				}
+				headRepositoryOwner {
+					login
+				}
+			}`;
+	}
+
+	query += `
+		}
+	}`;
+	return query;
+}
+
+async function fetchFromApi(owner, repo, numbers) {
+	const query = buildQuery(owner, repo, numbers);
+	const response = await api.v4(query);
+
+	if (response.data && response.data.repository) {
+		const pulls = {};
+		for (const [id, data] of Object.entries(response.data.repository)) {
+			pulls[id] = normalizePullInfo(owner, repo, data);
+		}
+		return pulls;
+	}
+}
+
+const normalizePullInfo = (owner, repo, data) => ({
```

Comment:
Why are these arrow functions rather than plain functions like the rest of the file?

Reply:
No particular reason. I can go either way.

Reply:
I prefer to use arrow functions only
- When used anonymously (not named, and not reused, so good for callbacks)
- To preserve `this`
- And some weird edge cases

And
```
function normalizePullInfo(owner, repo, data) {
```

is smaller and easier to interpret than
```
const normalizePullInfo = (owner, repo, data) => ({
```

Reply:
I suggested them that way because one was a one-liner. I usually prefer `function` because it's explicit (and usually shorter, except cases like this one)

Reply:
So, there is no standard in this project? I can change them all to plain function or arrow functions.

Reply:
No enforced standard at the moment, they are just whatever they happen to be and it doesn't make much of a difference, except personal preference (which varies)

Reply:
I kinda agree with @notlmn and change them to plain function.